promptsource/promptsource.py

```diff
@@ -55,13 +55,13 @@ def reset_template_state():
 
 def show_jinja(t, width=WIDTH):
     wrap = textwrap.fill(t, width=width, replace_whitespace=False)
-    out = highlight(wrap, DjangoLexer(), HtmlFormatter())
+    out = highlight(wrap, DjangoLexer(), HtmlFormatter(lineseparator="<br>"))
     st.write(out, unsafe_allow_html=True)
 
 
 def show_text(t, width=WIDTH):
     wrap = [textwrap.fill(subt, width=width, replace_whitespace=False) for subt in t.split("\n")]
-    wrap = "\n".join(wrap)
+    wrap = "<br>".join(wrap)
```

Comment:
If we use components with IDs instead of st.text()/write(), then we can use `st.markdown("<style> some_id : { white-space: pre-wrap; } </style>", unsafe_allow_html=True)` instead of the `<br>` here.
But I guess we don't want to modify too many places for now.

Reply:
I'm a bit confused. There are two question here right, what format to have for the yaml and how to print. But they are independent right?

Reply:
> There are two question here right, what format to have for the yaml and how to print. But they are independent right?

I was probably overthinking it but didn't explain what I considered explicitly, my apologies.
In short, I think we can ignore the yaml part and only be focused on UI representation. Unless #355 also wanted to have a unified yaml format. I suppose #355 doesn't really care about it.